### PR TITLE
Write metrics to CloudWatch

### PR DESCRIPTION
**Pull request by nathanleiby**
https://clever.atlassian.net/browse/PASSION-23

This is to help setup auto-scaling, triggered via CloudWatch alarms

The scaling metric  is based on  https://cloud.google.com/compute/docs/autoscaler/scaling-queue-based

Also moves from Godeps to Glide

### CODE REVIEW

**Comment by nathanleiby** [comment]
Most changes are in this commit: https://github.com/Clever/gearman-load-logger/pull/28/commits/28759d73141c1eb2d6c94ffd7219bcaaac464553

File: main.go [hunk 1]
```diff
@@ -16,20 +21,57 @@ const (
...
 			"running_workers":   status.Running,
 			"available_workers": status.AvailableWorkers,
 		})
+
+		// Write data to CloudWatch
+		jobToWorkerRatio := float64(status.Total) / float64(status.AvailableWorkers)
+		if math.IsInf(jobToWorkerRatio, 1) || math.IsNaN(jobToWorkerRatio) {
+			// handle divide-by-zero cases (x/0 or 0/0)
+			continue
+		}
```
Comment (xavi-) [inline main.go]: Could instead check if `status.AvailableWorkers` is 0?

**Review by xavi- — APPROVED** [review]
One small nit.  Otherwise lgtm

File: main.go (hunk 1 above)
Comment (nathanleiby) [reply main.go]: Simplified divide by zero check: 3b5c3561e72a33406b662250cb3240e429a2fbde

**Pull request merged**